pkg/types/powervs/validation/machinepool.go

```diff
@@ -0,0 +1,65 @@
+package validation
+
+import (
+	"github.com/google/uuid"
+	"github.com/openshift/installer/pkg/types/powervs"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"math"
+	"regexp"
+	"strconv"
+)
+
+// ValidateMachinePool checks that the specified machine pool is valid.
+func ValidateMachinePool(p *powervs.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Validate VolumeIDs
+	for i, volumeID := range p.VolumeIDs {
+		_, err := uuid.Parse(volumeID)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("volumeIDs").Index(i), volumeID, "Volume ID provided is not a UUID"))
+		}
+	}
+
+	// Validate Memory
+	if p.Memory != "" {
+		memory, err := strconv.ParseInt(p.Memory, 10, 64)
+		if err == nil {
+			if memory < 2 || memory > 64 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "Memory must be from 2 to 64 GB"))
+			}
+		} else {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "Memory must be a valid integer"))
```

Comment:
Be more precise about the message. Do not tell the user that it must be an integer, allow them to put in an integer outside of the prescribed bounds, and then later tell them the bounds.
```suggestion
			allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "memory must be an integer number of GB that is at least 2 and no more than 64"))
```